packages/editor/src/AppModel/FieldModel.ts

```diff
@@ -109,28 +137,90 @@ export class FieldModel implements IFieldModel {
       parseExpression(this.value as string).filter(exp => typeof exp !== 'string')
     );
 
+    this.refComponentInfos = {};
+    this.astNodes = {};
+
     exps.forEach(exp => {
       let lastIdentifier: ComponentId = '' as ComponentId;
-      simpleWalk((acornLoose as typeof acorn).parse(exp, { ecmaVersion: 2020 }), {
-        Expression: node => {
-          switch (node.type) {
+      const node = (acornLoose as typeof acorn).parse(exp, { ecmaVersion: 2020 });
+
+      this.astNodes[exp] = node as ASTNode;
+
+      simpleWalk(node, {
+        Expression: expressionNode => {
+          switch (expressionNode.type) {
             case 'Identifier':
-              const key = exp.slice(node.start, node.end) as ComponentId;
-              this.refs[key] = [];
+              const key = exp.slice(
+                expressionNode.start,
+                expressionNode.end
+              ) as ComponentId;
+
+              if (this.refComponentInfos[key]) {
+                this.refComponentInfos[key].componentIdASTNodes.push(
+                  expressionNode as ASTNode
+                );
+              } else {
+                this.refComponentInfos[key] = {
+                  componentIdASTNodes: [expressionNode as ASTNode],
+                  refProperties: [],
+                };
+              }
               lastIdentifier = key;
+
               break;
             case 'MemberExpression':
-              const str = exp.slice(node.start, node.end);
+              const str = exp.slice(expressionNode.start, expressionNode.end);
               let path = str.replace(lastIdentifier, '');
               if (path.startsWith('.')) {
                 path = path.slice(1, path.length);
               }
-              this.refs[lastIdentifier]?.push(path);
+              this.refComponentInfos[lastIdentifier]?.refProperties.push(path);
               break;
             default:
           }
         },
       });
     });
   }
+
+  private onReferenceIdChange({ oldId, newId }: AppModelEventType['idChange']) {
```

Comment:
This is the core part of the pr and it seems a little bit hard to understand. We should add some comments here to help us understand in the future.
Other things LGTM.